src/test/java/com/google/api/codegen/testdata/java_test_mock_service_impl_library.baseline

```diff
@@ -46,154 +47,280 @@ import java.util.List;
-   @Override
-   public void publishSeries(PublishSeriesRequest request,
-       StreamObserver<PublishSeriesResponse> responseObserver) {
-     requests.add(request);
-     responseObserver.onNext(PublishSeriesResponse.newBuilder().build());
-     responseObserver.onCompleted();
-   }
-
-   @Override
-   public void getBook(GetBookRequest request,
-       StreamObserver<Book> responseObserver) {
-     requests.add(request);
-     responseObserver.onNext(Book.newBuilder().build());
-     responseObserver.onCompleted();
-   }
-
-   @Override
-   public void listBooks(ListBooksRequest request,
-       StreamObserver<ListBooksResponse> responseObserver) {
-     requests.add(request);
-     responseObserver.onNext(ListBooksResponse.newBuilder().build());
-     responseObserver.onCompleted();
-   }
-
-   @Override
-   public void deleteBook(DeleteBookRequest request,
-       StreamObserver<Empty> responseObserver) {
-     requests.add(request);
-     responseObserver.onNext(Empty.newBuilder().build());
-     responseObserver.onCompleted();
-   }
-
-   @Override
-   public void updateBook(UpdateBookRequest request,
-       StreamObserver<Book> responseObserver) {
-     requests.add(request);
-     responseObserver.onNext(Book.newBuilder().build());
-     responseObserver.onCompleted();
-   }
-
-   @Override
-   public void moveBook(MoveBookRequest request,
-       StreamObserver<Book> responseObserver) {
-     requests.add(request);
-     responseObserver.onNext(Book.newBuilder().build());
-     responseObserver.onCompleted();
-   }
-
-   @Override
-   public void listStrings(ListStringsRequest request,
-       StreamObserver<ListStringsResponse> responseObserver) {
-     requests.add(request);
-     responseObserver.onNext(ListStringsResponse.newBuilder().build());
-     responseObserver.onCompleted();
-   }
-
-   @Override
-   public void addComments(AddCommentsRequest request,
-       StreamObserver<Empty> responseObserver) {
-     requests.add(request);
-     responseObserver.onNext(Empty.newBuilder().build());
-     responseObserver.onCompleted();
-   }
-
-   @Override
-   public void getBookFromArchive(GetBookFromArchiveRequest request,
-       StreamObserver<Book> responseObserver) {
-     requests.add(request);
-     responseObserver.onNext(Book.newBuilder().build());
-     responseObserver.onCompleted();
-   }
-
-   @Override
-   public void updateBookIndex(UpdateBookIndexRequest request,
-       StreamObserver<Empty> responseObserver) {
-     requests.add(request);
-     responseObserver.onNext(Empty.newBuilder().build());
-     responseObserver.onCompleted();
-   }
-
-   @Override
-   public void streamShelves(ListShelvesRequest request,
-       StreamObserver<ListShelvesResponse> responseObserver) {
-     requests.add(request);
-     responseObserver.onNext(ListShelvesResponse.newBuilder().build());
-     responseObserver.onCompleted();
-   }
+  private ArrayList<GeneratedMessage> requests;
+  private ArrayList<GeneratedMessage> responses;
```

Comment:
Nit: A LinkedList might be more efficient, since you only seem to ever pull the head off `responses`, and push a new head onto `requests`. The pattern is quite stack-like.